pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -141,12 +152,75 @@ func (is *InstanceService) GetAcceptableFloatingIP() (string, error) {
 	return "", fmt.Errorf("Don't have acceptable floating IP.")
 }
 
+// StructToString converts structs to strings --> returns an empty string if an error occurs
+func StructToString(opts *networks.ListOpts) string {
+	out, err := json.Marshal(opts)
+	if err != nil {
+		return ""
+	}
+
+	return string(out)
+}
+
+// A function for getting the id of a network by querying openstack with filters
+func getNetworkIDByFilters(is *InstanceService, opts *networks.ListOpts) (string, error) {
+	networkCount := 0
+	networkID := ""
+	if opts == nil {
+		return "", fmt.Errorf("Network filters must be provided")
+	}
+	pager := networks.List(is.networkClient, opts)
+	pager.EachPage(func(page pagination.Page) (bool, error) {
+		networkList, err := networks.ExtractNetworks(page)
```

Comment:
I think I would rather it fail if there are too many results, than choose one at random. At least that way, people know exactly what they are getting. Otherwise, that would be crazy confusing. But I can move the check up a bit.

Reply:
I didn't say you should choose one at random. I said, if there are more than one, then there are too many networks and you should fail. If there's just one, then you can access the first index and you can do away with the iteration below.

Reply:
Oh i misunderstood

Reply:
I think we could still do this